lib/data/dpa/topology.js

```diff
@@ -84,16 +80,19 @@ function updateWeight(obj) {
 // generate weight distribution
 function genWeightDistr(metadata, obj, depth) {
     if (!obj.leaf && obj.constructor === Object) {
+        const max = Object.keys(obj).map(val =>
+            obj[val].maxFragsNb).filter(nb => nb !== undefined);
         obj.wdistr = [{                                 // eslint-disable-line
             // ids: Object.keys(obj).map(key => obj[key].id).filter(id =>
             //         id !== undefined),
             ids: Object.keys(obj).filter(key =>
                 obj[key].constructor === Object),
             pdf: [],
             cdf: [],
-            max: Object.keys(obj).map(val =>
-                obj[val].maxFragsNb).filter(nb => nb !== undefined),
         }];
+        if (max.length > 0) {
+            obj.wdistr.max = max;                       // eslint-disable-line
```

Comment:
Maybe you can set ```obj.wdistr.max``` to ```0``` when you declare ```obj.wdistr```, so you do not add a property to the object just after.